includes/data-stores/class-wc-product-data-store-cpt.php

```diff
@@ -309,7 +309,7 @@ protected function read_visibility( &$product ) {
 	protected function read_attributes( &$product ) {
 		$meta_values = maybe_unserialize( get_post_meta( $product->get_id(), '_product_attributes', true ) );
 
-		if ( $meta_values ) {
+		if ( $meta_values && !empty( $meta_values ) ) {
```

Comment:
If you have an empty check, the original check for `$meta_values` on this line can be removed. I'd also suggest adding a second check for `is_array`.

Have a read of https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/ too - you need to leave a space after the `!`